src/server/debug.rs

```diff
@@ -444,6 +482,223 @@ impl Debugger {
+        loop {
+            if next_default {
+                default = self.next_default(key)?;
+                next_default = false;
+            }
+            if next_write {
+                write = self.next_write(key)?;
+                next_write = false;
+            }
+            if next_lock {
+                lock = self.next_lock(key)?;
+                next_lock = false;
+            }
+
+            // If lock exists, check whether the records in DEFAULT and WRITE
+            // match it.
+            if let Some(ref l) = lock {
+                // All write records' ts should be less than lock's ts.
+                if let Some((commit_ts, _)) = write {
+                    if l.ts <= commit_ts {
+                        warn!(
+                            "LOCK ts is less than WRITE ts, key: {}, lock_ts: {}, commit_ts: {}",
+                            escape(key),
+                            l.ts,
+                            commit_ts
+                        );
+                        self.delete(wb, CF_LOCK, key, None)?;
+                        self.lock_fix_count += 1;
+                        next_lock = true;
+                        continue;
+                    }
+                }
+
+                // If the lock's type is PUT and contains no short value, there
+                // should be a corresponding default record.
+                if l.lock_type == LockType::Put && l.short_value.is_none() {
+                    match default {
+                        Some(start_ts) if start_ts == l.ts => {
+                            default = self.next_default(key)?;
+                        }
+                        _ => {
+                            warn!(
+                                "no corresponding DEFAULT record for LOCK, key: {}, lock_ts: {}",
+                                escape(key),
+                                l.ts
+                            );
+                            self.delete(wb, CF_LOCK, key, None)?;
+                            self.lock_fix_count += 1;
+                        }
+                    }
+                }
+                next_lock = true;
+                continue;
+            }
+
+            // For none-put write or write with short_value, no DEFAULT record
+            // is needed.
+            if let Some((_, ref w)) = write {
+                if w.write_type != WriteType::Put || w.short_value.is_some() {
+                    next_write = true;
+                    continue;
+                }
+            }
+
+            // The start_ts of DEFAULT and WRITE should be matched.
+            match (default, &write) {
+                (Some(start_ts), &Some((_, ref w))) if start_ts == w.start_ts => {
+                    next_default = true;
+                    next_write = true;
+                    continue;
+                }
+                (Some(start_ts), &Some((_, ref w))) if start_ts < w.start_ts => next_write = true,
```

Comment:
When

start_ts | w.start_ts
---- | ----
2 | 2
1 | None

The first default start_ts = 1, write start_ts = 2, this will delete the write?

Reply:
Oh, seems my mistake, the keys are ordered descendingly by the timestamp?